src/core/geometry/qgsgeometry.cpp

```diff
@@ -259,7 +263,7 @@ QgsGeometry QgsGeometry::fromBox3D( const QgsBox3D &box )
     return fromRect( box.toRectangle() );
   }
 
-  std::unique_ptr< QgsMultiPolygon > multiPolygon = std::make_unique< QgsMultiPolygon >();
+  std::unique_ptr< QgsPolyhedralSurface > polyhedralSurface = std::make_unique< QgsPolyhedralSurface >();
```

Comment:
I think this behaviour should be noted in the docs for fromBox3D -- eg "\returns a 2D polygon geometry if the box is purely 2d, otherwise returns a polyhedral surface geometry."

Reply:
Done.